src/callkit.cs

```diff
@@ -0,0 +1,509 @@
+
+		[Async]
+		[Export ("reloadExtensionWithIdentifier:completionHandler:")]
+		void ReloadExtension (string identifier, [NullAllowed] Action<NSError> completion);
+
+		[Async]
+		[Export ("getEnabledStatusForExtensionWithIdentifier:completionHandler:")]
+		void GetEnabledStatusForExtension (string identifier, Action<CXCallDirectoryEnabledStatus, NSError> completion);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[BaseType (typeof (NSObject))]
+	interface CXCallDirectoryProvider {
+
+		[Export ("beginRequestWithExtensionContext:")]
+		void BeginRequest (CXCallDirectoryExtensionContext context);
+	}
+
+	interface ICXCallObserverDelegate { }
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[Protocol, Model]
+	[BaseType (typeof (NSObject))]
+	interface CXCallObserverDelegate {
+
+		[Abstract]
+		[Export ("callObserver:callChanged:")]
+		void CallChanged (CXCallObserver callObserver, CXCall call);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[BaseType (typeof (NSObject))]
+	interface CXCallObserver {
+
+		[Export ("calls", ArgumentSemantic.Copy)]
+		CXCall [] Calls { get; }
+
+		[Export ("setDelegate:queue:")]
+		void SetDelegate ([NullAllowed] ICXCallObserverDelegate aDelegate, [NullAllowed] DispatchQueue queue);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[BaseType (typeof (NSObject))]
+	interface CXCallUpdate : NSCopying {
+
+		[NullAllowed, Export ("callerIdentifier")]
+		string CallerIdentifier { get; set; }
+
+		[NullAllowed, Export ("localizedCallerName")]
+		string LocalizedCallerName { get; set; }
+
+		[Export ("supportsHolding", ArgumentSemantic.Assign)]
+		bool SupportsHolding { get; set; }
+
+		[Export ("supportsGrouping", ArgumentSemantic.Assign)]
+		bool SupportsGrouping { get; set; }
+
+		[Export ("supportsUngrouping", ArgumentSemantic.Assign)]
+		bool SupportsUngrouping { get; set; }
+
+		[Export ("supportsDTMF", ArgumentSemantic.Assign)]
+		bool SupportsDtmf { get; set; }
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[BaseType (typeof (CXCallAction))]
+	interface CXEndCallAction {
+
+		[Export ("fulfillWithDateEnded:")]
+		void Fulfill (NSDate dateEnded);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[BaseType (typeof (CXCallAction), Name = "CXPlayDTMFCallAction")]
+	interface CXPlayDtmfCallAction {
+
+		[Export ("digits")]
+		string Digits { get; set; }
+
+		[Export ("type", ArgumentSemantic.Assign)]
+		CXPlayDtmfCallActionType Type { get; set; }
+	}
+
+	interface ICXProviderDelegate { }
+
+	[Protocol, Model]
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.MacOSX, 10, 12)]
+	[BaseType (typeof (NSObject))]
+	interface CXProviderDelegate {
+
+		[Export ("providerDidReset:")]
+		void ProviderDidReset (CXProvider provider);
```

Comment:
Should not start with `Provider`, the only reason if differs from (most) others is because it's a single parameter method (so the `:` must be at the end) while the others can have their `:` after the `provider`.